### PR TITLE
ciao-image: Add delete image support

### PR DESCRIPTION
**Pull request by obedmr**
This commit adds support for deleting images. Now, you will be able to
delete images with the command:
```
  ciao-cli image delete -image 78306709-9720-417e-97c6-7b941cb5565b
```
If the operation was successful, you will get a 204 HTTP return code.
If not, it may return the proper error, for example a 404 error could
come when the image does not exist.
Signed-off-by: Munoz, Obed N <obed.n.munoz@intel.com>

### CODE REVIEW

**Comment by markdryan** [comment]
@obedmr Code looks okay to me.  Could you add a description to your commit message?

@sameo, @kaccardi is this one okay for you?

File: openstack/image/api.go [hunk 1]
```diff
@@ -362,6 +378,19 @@ func uploadImage(context *Context, w http.ResponseWriter, r *http.Request) (APIR
...
+
+	defer r.Body.Close()
+
+	resp, err := context.DeleteImage(imageID)
+	if err != nil {
+		return errorResponse(err), err
+	}
+	return APIResponse{http.StatusNoContent, resp}, nil
+}
+
```
Comment (kaccardi) [inline openstack/image/api.go]: You are returning StatusNoContent - but you are returning a response. StatusNoContent should be used when there is no response, and the openstack spec says that /v2/images/{image_id} DELETE should return no response. http://developer.openstack.org/api-ref/image/v2/index.html?expanded=delete-an-image-detail#delete-an-image

File: openstack/image/api.go [hunk 2]
```diff
@@ -201,9 +214,10 @@ type APIConfig struct {
 // information needed to implement the image endpoints.
 type Service interface {
 	CreateImage(CreateImageRequest) (DefaultResponse, error)
-	UploadImage(string, io.Reader) (UploadImageResponse, error)
+	UploadImage(string, io.Reader) (NoContentImageResponse, error)
 	ListImages() ([]DefaultResponse, error)
```
Comment (kaccardi) [inline openstack/image/api.go]: I know this isn't your bug - but I see that UploadImage also returns a response, when it should also return no response as indicated by the api spec.
http://developer.openstack.org/api-ref/image/v2/index.html?expanded=delete-an-image-detail,upload-binary-image-data-detail#upload-binary-image-data

File: openstack/image/api.go (hunk 1 above)
Comment (obedmr) [inline openstack/image/api.go]: @kaccardi, you're right, good catch. /me updating it

File: openstack/image/api.go (hunk 2 above)
Comment (obedmr) [inline openstack/image/api.go]: /me fixing it

**Comment by obedmr** [comment]
@kaccardi, @markdryan @sameo, it's updated. When you have a chance could you take a look on it?

File: openstack/image/api.go [hunk 3]
```diff
@@ -253,6 +267,8 @@ func (h APIHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 // on return values all the time.
 func errorResponse(err error) APIResponse {
 	switch err {
+	case ErrNoImage:
+		return APIResponse{http.StatusNotFound, nil}
 	default:
```
Comment (kaccardi) [inline openstack/image/api.go]: Shouldn't there be a response for ErrImageSaving? I would think this would be a 4xx response and not a 5xx.

Comment (obedmr) [inline openstack/image/api.go]: @kaccardi, this error is going to be used when you're using the Get method. If the image is not Found, it's getting a 404 error, which is the http.StatusNotFound 
https://golang.org/pkg/net/http/
```
  (StatusNotFound                     = 404 // RFC 7231, 6.5.4)
```

The idea of this errorResponse function (AFAIK) is to be the one for handling the different HTTP errors. For now those are the implemented ones, expect more to be added.

Comment (kaccardi) [inline openstack/image/api.go]: @obedmr ErrImageSaving is returned from UploadImage - which is a PUT method. This error is encountered if the state of the image is already in "Saving" when someone tries to do another upload. Correct?

Comment (obedmr) [inline openstack/image/api.go]: yes, this case should be handled differently.

**Comment by obedmr** [comment]
@kaccardi, when you have a chance, could you take a look on it? I just added the missing test.

**Pull request merged**